### PR TITLE
Closing Bookmark details dialog with X should also refresh list entry

### LINKED ISSUE #12387: Dog-ear mark on the current page is not updated after exiting the page browser
* KOReader version: 2024.07-90
* Device: Kobo Forma, Desktop

#### Issue

After exiting the page browser (via the top-right X), the dog-ear mark on the current page is not properly updated if a bookmark has been added/removed from within the browser.

#### Steps to reproduce

- Open the page browser
- Tap-and-hold on the current page to toggle a bookmark
- Exit the page browser (via the top-right X)

At this point, a bookmark for the current page has indeed been toggled, but the dog-ear mark shows the opposite.

### LINKED ISSUE #12389: Closing Bookmark details dialog with top-right X does not properly refresh list entry
* KOReader version: v2024.07-118
* Device: Desktop

#### Issue

Closing the Bookmark details dialog with the top-right X, unlike the "Close" button, does not refresh the corresponding bookmark list entry if the details were modified.

#### Steps to reproduce

(Assuming the current book has at least one bookmark/highlight.)

- Open the bookmarks list
- Tap-and-hold on one entry
- Either select to add a note or edit the current one
- Write something and select Save
- Close the Bookmark details dialog with the top-right X

At this point, the bookmark details were successfully modified, but the list entry does not reflect this.

### PR DESCRIPTION
**Pull request by fbriere**
Fixes #12389
Fixes #12387

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/12390)

### CODE REVIEW

**Comment by fbriere** [comment]
(I'm not sure why my first attempt worked just fine even though `item_table` is supposed to be a local variable that's out of scope here.  Oh well.)

File: frontend/apps/reader/modules/readerbookmark.lua [hunk 1]
```diff
@@ -1202,6 +1196,23 @@ function ReaderBookmark:showBookmarkDetails(item)
     return true
 end
 
+-- Refresh the bookmark list whenever details may have been edited
+function ReaderBookmark:refreshBookmarkDetails()
```
Comment (hius07) [inline frontend/apps/reader/modules/readerbookmark.lua:1200]: I'd call it `_updateBookmarkList` and set a local function withinin `showBookmarkDetails()` as it is not used outside.

File: frontend/apps/reader/modules/readerbookmark.lua [hunk 2]
```diff
@@ -1095,6 +1095,10 @@ function ReaderBookmark:showBookmarkDetails(item)
         title = T(_("Bookmark details (%1/%2)"), item.idx, #item_table),
         text = text,
         text_type = "bookmark",
+        close_callback = function()
+            self:refreshBookmarkDetails()
+            UIManager:close(textviewer)
```
Comment (hius07) [inline frontend/apps/reader/modules/readerbookmark.lua:1100]: Not needed, already closed in
https://github.com/koreader/koreader/blob/698fc35c3f1a10233cf69085a87fb912e2f79841/frontend/ui/widget/textviewer.lua#L384

**Review by hius07 — APPROVED** [review]

**Comment by poire-z** [comment]
Can you do the same as last commit in bookmapwidget.lua ?
One can launch Book map, tap and get a Pagebrowser, toggle bookmark, close Pagebrowser, close Bookmap, and they should see the dogear change.

**Comment by fbriere** [comment]
> Can you do the same as last commit in bookmapwidget.lua ?

Done.

**Pull request merged**